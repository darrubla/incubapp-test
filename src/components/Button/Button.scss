@import '../../generals.scss';

.Button {
  display: flex;
  border: none;
  position: relative;
  align-items: center;
  justify-content: center;
  border-radius: $spacing-s;
  padding: 10px;
  margin-bottom: 10px;
  color: white;
  border: groove;
  cursor: pointer;
  @extend .fn-body-1-semi-bold;

  &:disabled {
    cursor: not-allowed;
  }

  &.no-behavior {
    user-select: none;

    &:hover {
      cursor: default;
    }
  }

  &[primary] {
    background-color: $color-main;
    color: $color-white;

    &:hover {
      background-color: $main-hover;
    }

    &:active {
      background-color: $main-active;
    }

    &:disabled {
      background-color: $main-disabled;
    }
  }

  &[secondary] {
    background-color: $color-light-gray;
    color: $color-white;
    box-sizing: border-box;

    &:hover {
      background-color: $color-dark-gray
    }

    &:active {
      color: $main-active;
    }

    &:disabled {
      color: $main-disabled;
      border: 1px solid $main-disabled;
    }

  }

  & svg {
    position: absolute;
    left: 12px;
  }
}