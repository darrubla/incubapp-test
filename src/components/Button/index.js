import React from 'react'
import PropTypes from 'prop-types'

import { Button } from '@mui/material'

import './Button.scss'

function ButtonComponent(props) {
  const { className, id, text, disabled, secondary, action, children, name } =
    props
  return (
    <Button
      type="button"
      name={name}
      className={`Button ${className}-Button`}
      disabled={disabled}
      id={id}
      primary={!secondary ? 'true' : undefined}
      secondary={secondary ? 'true' : undefined}
      onClick={action}
    >
      {children}
      {text}
    </Button>
  )
}

ButtonComponent.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  text: PropTypes.string,
  disabled: PropTypes.bool,
  secondary: PropTypes.bool,
  action: PropTypes.func,
  children: PropTypes.node,
  name: PropTypes.string,
}

ButtonComponent.defaultProps = {
  className: '',
  text: 'Click me',
  disabled: false,
  secondary: false,
  action: () => {},
  children: undefined,
  name: '',
}

export default ButtonComponent
