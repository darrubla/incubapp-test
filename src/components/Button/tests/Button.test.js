import React from 'react'
import { render } from '@testing-library/react'

import Button from '../index'

describe('<Button />', () => {
  it('Should render properly', () => {
    const props = {
      id: 'test button',
      action: jest.fn(),
      text: 'test',
    }
    const { getByText } = render(<Button {...props} />)
    expect(getByText('test')).toBeTruthy()
  })

  it('Should render and match the snapshot', () => {
    const props = {
      id: 'test button',
      action: jest.fn(),
      text: 'test',
    }
    const {
      container: { firstChild },
    } = render(<Button {...props} />)
    expect(firstChild).toMatchSnapshot()
  })
})
