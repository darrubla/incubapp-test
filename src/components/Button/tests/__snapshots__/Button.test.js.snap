// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render and match the snapshot 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Button -Button css-1e6y48t-MuiButtonBase-root-MuiButton-root"
  id="test button"
  name=""
  primary="true"
  tabindex="0"
  type="button"
>
  test
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
